src/Sparrow/Json/AsyncBlittableJsonTextWriter.cs

```diff
@@ -46,7 +46,9 @@ public async ValueTask<int> FlushAsync(CancellationToken token = default)
         public async ValueTask DisposeAsync()
         {
             DisposeInternal();
-            await FlushAsync().ConfigureAwait(false);
+            if (await FlushAsync().ConfigureAwait(false) > 0)
+                await _outputStream.FlushAsync().ConfigureAwait(false); // flush the output stream underlying stream as well
```

Comment:
why this was added? this is called during the dispose so I think dispose of that stream will do the flash? have you encountered an issue?